server/services/mappers/alertMapper.ts

```diff
@@ -140,3 +147,46 @@ export const toAlertFlagLabels = (alerts: Alert[], alertFlags: AlertFlagLabel[])
     })
     .filter(alert => alert.alertIds?.length)
 }
+
+export const toAlertTypesFilters = (alerts: Alert[]) => {
+  return alerts.reduce(
+    (acc, alert) => {
+      if (alert.isActive) {
+        acc.activeAlertTypesFilter = {
+          ...acc.activeAlertTypesFilter,
+          [alert.alertCode.alertTypeCode]: {
+            code: alert.alertCode.alertTypeCode,
+            description: alert.alertCode.alertTypeDescription,
+            count: acc.activeAlertTypesFilter[alert.alertCode.alertTypeCode]
+              ? acc.activeAlertTypesFilter[alert.alertCode.alertTypeCode].count + 1
+              : 1,
+            checked: false,
+          },
+        }
+      } else {
+        acc.inactiveAlertTypesFilter = {
+          ...acc.inactiveAlertTypesFilter,
+          [alert.alertCode.alertTypeCode]: {
+            code: alert.alertCode.alertTypeCode,
+            description: alert.alertCode.alertTypeDescription,
+            count: acc.inactiveAlertTypesFilter[alert.alertCode.alertTypeCode]
+              ? acc.inactiveAlertTypesFilter[alert.alertCode.alertTypeCode].count + 1
+              : 1,
+            checked: false,
+          },
+        }
+      }
+      return acc
+    },
+    { activeAlertTypesFilter: {}, inactiveAlertTypesFilter: {} },
```

Comment:
Given these two blocks of code (one for active and one for inactive) are basically the same, would it be more readable to convert this to:

```
export const toAlertTypesFilter = ({ alerts, active }: { alerts: Alert[]; active: boolean }) => {
  return alerts
    .filter(alert => alert.isActive === active)
    .reduce((acc, alert) => {
      return {
        ...acc,
        [alert.alertCode.alertTypeCode]: {
          code: alert.alertCode.alertTypeCode,
          description: alert.alertCode.alertTypeDescription,
          count: (acc[alert.alertCode.alertTypeCode] ?? 0) + 1,
          checked: false,
        },
      }
    }, {})
}

export const toAlertSummaryData = (alerts: Alert[], alertFlags: AlertFlagLabel[]): AlertSummaryData => {
  return {
    activeAlertCount: alerts.filter(alert => alert.isActive).length,
    inactiveAlertCount: alerts.filter(alert => !alert.isActive).length,
    activeAlertTypesFilter: toAlertTypesFilter({ alerts, active: true }),
    inactiveAlertTypesFilter: toAlertTypesFilter({ alerts, active: false }),
    alertFlags: toAlertFlagLabels(alerts, alertFlags),
  }
}
```

Reply:
(So call the method twice, to get the active and inactive)